omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,205 @@
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// this is the proper sequence to set nonCodedAllergen
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, \"nonCodedAllergen\": \"test non coded\" }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("test non coded", Util.getByPath(newAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesForNonExistingPatientUsingEmptyAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertNotEquals(Allergies.NO_KNOWN_ALLERGIES, allergies.getAllergyStatus());
+
+		String nonExistentPatientURI = "patient/nonExistingPatient/allergy";
+		String json = "{}";
+		handle(newPutRequest(nonExistentPatientURI, json));
+		
+		allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertNotEquals(Allergies.NO_KNOWN_ALLERGIES, allergies.getAllergyStatus());
+	}
+	
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingNonEmptyAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertNotEquals(Allergies.NO_KNOWN_ALLERGIES, allergies.getAllergyStatus());
+
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"nonCodedAllergen\": \"test non coded\" }}";
+		handle(newPutRequest(getURI(), json));
+		
+		allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertNotEquals(Allergies.NO_KNOWN_ALLERGIES, allergies.getAllergyStatus());
+	}
+	
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingEmptyAllergies() throws Exception

```

Comment:
The method names seems to suggest that it should throw an exception whenever one attempts to set no know allergies for any existing patient. Is this correct?